File: GeeksForGeeks-Problems/DSA/Mathematics/Quadratic Equation Roots.cpp
// User function Template for C++

class Solution
{
public:
    vector<int> quadraticRoots(int a, int b, int c)
    {
        // code here
        int D = (b * b) - 4 * a * c;
        if (D < 0)
        {
            return {-1};
        }
        else
        {
            int x = floor(((-b) + sqrt(D)) / (2 * a));
            int y = floor(((-b) - sqrt(D)) / (2 * a));
            if (x > y)
            {
                return {x, y};
            }
            else
            {
                return {y, x};
            }
        }
    }
};

File: GeeksForGeeks-Problems/DSA/Mathematics/Quadratic Equation Roots.java
class Solution
{
public
    ArrayList<Integer> quadraticRoots(int a, int b, int c)
    {
        // code here
        ArrayList<Integer> ans = new ArrayList<Integer>();
        double d = (b * b) - (4 * a * c);
        if (d < 0)
        {
            ans.add(-1);
            return ans;
        }
        else
        {
            double l = Math.sqrt(d);
            double a1 = (-b + l) / (2.0 * a);
            double a2 = (-b - l) / (2.0 * a);
            int r1 = (int)Math.floor(a1);
            int r2 = (int)Math.floor(a2);
            if (r1 < r2)
            {
                ans.add(r2);
                ans.add(r1);
            }
            else
            {
                ans.add(r1);
                ans.add(r2);
            }
        }
        return ans;
    }
}